backend/src/admin.service.ts

```diff
@@ -85,40 +98,49 @@ export class AdminService {
       }
 
       // parse planning and execution time from query plan
-      if (analysis != null) {
-        submission.planning_time = parseFloat(analysis[analysis.length - 2]["QUERY PLAN"].substring(15, 20));
-        submission.execution_time = parseFloat(analysis[analysis.length - 1]["QUERY PLAN"].substring(16, 21));
+      if (planning_times.length > 0) { 
+        const average_planning_time = planning_times.reduce((a, b) => a + b) / planning_times.length;
+        submission.planning_time = parseFloat(average_planning_time.toFixed(2));
+      }
+      
+      if (execution_times.length > 0) {
+        const average_execution_time = execution_times.reduce((a, b) => a + b) / execution_times.length;
+        submission.execution_time = parseFloat(average_execution_time.toFixed(2));
```

Comment:
do we want to care in the case that `execution_times.length < run_count`? means it sometimes fails and sometimes returns a result? i know it's unlikely